### ISSUE TITLE
LDAP Group Filter and Group Base DN have no affect

### ISSUE THREAD

**Issue by MnrGreg** [goharbor/harbor#12613]
**Expected behavior and actual behavior:**
When a user authenticates against Harbor LDAP backend, one expects Harbor to lookup the associated LDAP groups and create entries for each, based on the filtering and naming configured through:

-   ldap_group_attribute_name
-   ldap_group_base_dn
-   ldap_group_search_filter

What actually happens is that Harbor ignores the `ldap_group_base_dn`; the `ldap_group_search_filter` and creates all of the groups associated with the user in the database. It also sets the group names to that of the DN attribute and not what is specified through `ldap_group_attribute_name`.

**Steps to reproduce the problem:**

1.  Set LDAP Group configurations to:
```yaml
  "ldap_group_attribute_name": "cn",
  "ldap_group_base_dn": "OU=Containers,OU=Resources,DC=redacted,DC=com",
  "ldap_group_search_filter": "(&(objectclass=group)(cn=APR_CONTAINERS*))",
```

2. Authenticate a user

3. Review created group entries.
```shell
registry=# select * from user_group;
 id  |                                      group_name                                       | group_type |                                     ldap_group_dn                                     |       creation_time        |        update_time         
-----+---------------------------------------------------------------------------------------+------------+---------------------------------------------------------------------------------------+----------------------------+----------------------------
 540 | CN=APR_CyberArk_TSHypervisorServices,OU=CyberArk,OU=Resources,DC=redacted,DC=com          |          1 | cn=apr_cyberark_tshypervisorservices,ou=cyberark,ou=resources,dc=redacted,dc=com          | 2020-07-28 19:11:25.723819 | 2020-07-28 19:11:25.723819
 541 | CN=APR_CyberArk_TSHypervisorServicesApproval,OU=CyberArk,OU=Resources,DC=redacted,DC=com  |          1 | cn=apr_cyberark_tshypervisorservicesapproval,ou=cyberark,ou=resources,dc=redacted,dc=com  | 2020-07-28 19:11:25.725605 | 2020-07-28 19:11:25.725605
```

A. Based on search filter and base DN neither of the above groups should have been created. 
B. Had they matched the filter and base DN, they should have been created with the CN value, not the DN value.

**Versions:**
Please specify the versions of following systems.
- harbor version: [2.0.1]

**Comment by MnrGreg** [comment]
PR https://github.com/goharbor/harbor/pull/11060 fixes this issue by correctly using the `ldap_group_attribute_name` for the Group Name during user logon.

Also `ldap_group_search_filter` does seem to work when matching a single key:value and not a more complicated multi-query filter. e.g. :
```yaml
 "ldap_group_search_filter": "objectclass=group",
```


`ldap_group_base_dn` still does not seem to apply when creating groups during user logon.

**Comment by stonezdj** [comment]
For the issue B. Had they matched the filter and base DN, they should have been created with the CN value, not the DN value. it is fixed in Harbor 2.0.x with PR: https://github.com/goharbor/harbor/pull/11060

**Comment by alex1989hu** [comment]
/cc @stonezdj 

# Original
Works with `v2.0.2` and latter releases of `v1.x.x`.

`memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe`

**Result**: I can do log in, I have my permission but I see these in `core` logs:

```
2020-09-21T07:27:36Z [ERROR] [/common/utils/ldap/ldap.go:302]: Wrong filter format, filter:()                                                                                                                    │
2020-09-21T07:27:36Z [WARNING] [/core/auth/ldap/ldap.go:117]: Can not get the ldap group name with DN CN=<redacted-1>, error invalid filter syntax
2020-09-21T07:27:36Z [WARNING] [/core/auth/ldap/ldap.go:117]: Can not get the ldap group name with DN CN=<redacted-2>, error invalid filter syntax
...
```

## LDAP Filter v2

Lets modify the `LDAP Filter`, mod: `'....'`

`'memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe'`

**Result:** I can do log in, I have my permission but I see the same error and warming messages *(see original)* in `core` logs.

## LDAP Filter v3

Lets modify the `LDAP Filter`, mod: `'( .... )'`


`'(memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe)'`

**Result:**  I can not do a login, `core` logs:

```
2020-09-21T07:35:34Z [WARNING] [/core/auth/ldap/ldap.go:65]: ldap search fail: invalid filter syntax                                                                                                             │
2020-09-21T07:35:34Z [ERROR] [/core/controllers/base.go:109]: Error occurred in UserLogin: invalid filter syntax
```

=> **KO**
